source/unicodeScriptData.py

```diff
@@ -0,0 +1,824 @@
+scriptCode= [
```

Comment:
Please add the usual copyright header. Also a docstring / explanation for this would be helpful.
Include in the docstring answers to the following:
- What each value represents, is it (unicodeStartOfRange, unicodeEndOfRange, rangeDescription)?
- Should value ranges overlap?
- Should all values be represented?
- Are these values contiguous?

Please also add unit tests to confirm the above restrictions should they exists.

Reply:
Added copyright and doc string with details about each item, but adding test cases may not be needed as these values are obtained from scripts.txt and it is done occasionally.

Reply:
Even though this data is created programmatically, tests for the data will help to ensure its correctness. It will be hard to test the script that creates this data, and it will complicate the code in NVDA to remove the assumptions made. An automated test to ensure that these assumptions are correct will make it much easier to track down the kind of bugs that would arise if this data does get out of order / values overlapping.

Please also state if the start / end of each range is inclusive or exclusive.

Reply:
This range comes from scripts.txt which comes from unicode.org. Should we still check it? I am assuming and I could be wrong that Unicode would not make an overlapping list.

Reply:
Another point, this list is sorted and is created once in a while.

Reply:
I'm more concerned that a mistake is made in the process of creating this. Perhaps the source data on unicode.org changes format and this causes a bug in the importer, this is even more likely that a mistake will be made if its been a while since someone last went through the process, and there is nothing in the production code to ensure the assumptions are met. It's quite hard to manually check that our assumptions about the `scriptCode` data hold true, but it's quite easy to write a unit test to do the same.

Reply:
Added 3 tests for checking the validity of the list.
1. test_unicodeRangesEntryStartLessEqualEnd
2. test_unicodeRangesEntriesDoNotOverlapAndAreSorted 
3. test_unicodeRangesEntryScriptNamesExist

Do let me know if we need to add more tests.

Reply:
For completeness, we could also test that the first elements start value is greater than 0.